### PR TITLE
Adding new image facts module

### PR DESCRIPTION
**Pull request by dagnello**

### CODE REVIEW

**Comment by dagnello** [comment]
cc @bcoca @emonty @Shrews @j2sol

File: cloud/openstack/os_image_facts.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this software.  If not, see <http://www.gnu.org/licenses/>.
+
+try:
+    import shade
+    from shade import meta
```
Comment (omgjlk) [inline cloud/openstack/os_image_facts.py]: `meta` is unused, please remove this import.

File: cloud/openstack/os_image_facts.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    HAS_SHADE = True
+except ImportError:
+    HAS_SHADE = False
+
+DOCUMENTATION = '''
+---
+module: os_image_facts
+short_description: Retrieve facts about a compute instance
```
Comment (omgjlk) [inline cloud/openstack/os_image_facts.py]: This description doesn't match the module's intended function.

File: cloud/openstack/os_image_facts.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+version_added: "2.0"
+author: "Davide Agnello (@dagnello)"
+description:
+    - Retrieve facts about a image instance from OpenStack.
```
Comment (omgjlk) [inline cloud/openstack/os_image_facts.py]: `instance` is an overloaded term. I would use a different term, or just `Retrieve facts regarding an image within OpenStack`

**Comment by omgjlk** [comment]
Return data should be documented.

**Comment by dagnello** [comment]
addressing comments.

**Comment by dagnello** [comment]
@bcoca @emonty @Shrews @j2sol ready

File: cloud/openstack/os_image_facts.py [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+EXAMPLES = '''
+# Gather facts about a previously created image named image1
+- os_networks_facts:
+    auth:
+      auth_url: https://your_api_url.com:9000/v2.0
+      username: user
+      password: password
+      project_name: someproject
+    image: image1
+- debug: var=openstack
```
Comment (wimnat) [inline cloud/openstack/os_image_facts.py:54]: The example doesn't match the module name or parameters

**Comment by dagnello** [comment]
@bcoca @emonty @Shrews @j2sol all comments have been addressed.

**Comment by omgjlk** [comment]
shipit

**Comment by gregdek** [comment]
Thanks @dagnello for the module, and thanks @j2sol for the review.

Marking for potential inclusion. cc: @ansible/core team for a final look.

**Comment by dagnello** [comment]
@gregdek thank you

**Pull request merged**